community/security/src/main/java/org/neo4j/server/security/auth/User.java

```diff
@@ -94,6 +104,7 @@ public boolean equals( Object o )
     public int hashCode()
     {
         int result = name != null ? name.hashCode() : 0;
+        result = 31 * result + ( group != null ? group.hashCode() : 0);
```

Comment:
Equivalence of User _objects_ is only used to detect concurrent modification of the auth repository  (throwing `ConcurrentModificationException`), so this `hashCode` just matches `equals`.
Whenever the user identity is compared it is by `user.name()`.